contrib/rpm/polybar.spec

```diff
@@ -1,70 +1,96 @@
 #
 # spec file for package polybar
-# Initially created for openSUSE
+# Created for openSUSE and Fedora
 #
+Name:          polybar	
+Version:       3.3.0
+Release:       1%{?dist}
+Summary:       helps users build beautiful and highly customizable status bars
 
-Name:           polybar
-Version:        3.3.0
-Release:        0
-Summary:        A fast and easy-to-use status bar
-License:        MIT
-Group:          System/GUI/Other
-URL:            https://github.com/jaagr/polybar
-Source:         https://github.com/jaagr/polybar/archive/%{version}.tar.gz
-BuildRequires:  clang >= 3.4
-BuildRequires:  cmake >= 3.1
-BuildRequires:  pkgconfig
-BuildRequires:  python-xml
-BuildRequires:  xcb-util-image-devel
-BuildRequires:  xcb-util-wm-devel
-# optional dependency
-BuildRequires:  pkgconfig(alsa)
-# main dependency
-BuildRequires:  pkgconfig(cairo)
-BuildRequires:  pkgconfig(jsoncpp)
-BuildRequires:  pkgconfig(libcurl)
-BuildRequires:  pkgconfig(libpulse)
-BuildRequires:  pkgconfig(python)
-BuildRequires:  pkgconfig(xcb)
-BuildRequires:  pkgconfig(xcb-proto)
-BuildRequires:  pkgconfig(xcb-util)
+Group:         System/GUI/Other
+License:       MIT
+URL:           https://github.com/jaagr/%{name}	
+Source0:       %{url}/releases/download/%{version}/%{name}-%{version}.tar
+
+BuildRequires: clang >= 3.4
+BuildRequires: cmake >= 3.1
+BuildRequires: git
+BuildRequires: pkgconfig(cairo)
+BuildRequires: pkgconfig(xcb)
+BuildRequires: pkgconfig(xcb-util)
+BuildRequires: pkgconfig(xcb-proto)
+BuildRequires: pkgconfig(xcb-image)
+BuildRequires: pkgconfig(xcb-ewmh)
+BuildRequires: pkgconfig(xcb-xrm)
+BuildRequires: pkgconfig(xcb-cursor)
+BuildRequires: pkgconfig(alsa)
+BuildRequires: pkgconfig(libpulse)
+BuildRequires: pkgconfig(jsoncpp)
+BuildRequires: pkgconfig(libcurl)
+BuildRequires: pkgconfig(libmpdclient)
+BuildRequires: pkgconfig(libnl-genl-3.0)
+
+%if 0%{?fedora}
+BuildRequires: ninja-build
+BuildRequires: python2
+BuildRequires: wireless-tools-devel
+BuildRequires: i3-ipc
+%else
+BuildRequires: ninja
+BuildRequires: pkgconfig
+BuildRequires: python-xml
+BuildRequires: pkgconfig(python)
 %if 0%{?suse_version} <= 1315
-BuildRequires:  i3-devel
+BuildRequires: i3-devel
 %else
-BuildRequires:  i3-gaps-devel
+BuildRequires: i3-gaps-devel
 %endif
 %if 0%{?suse_version}
-BuildRequires:  libiw-devel
+BuildRequires: libiw-devel
 %else
-BuildRequires:  wireless-tools-devel
+BuildRequires: wireless-tools-devel
+%endif
 %endif
 
 %description
-A fast and easy-to-use status bar for tilling WM
+Polybar aims to help users build beautiful and highly customizable status bars for their desktop environment, without the need of having a black belt in shell scripting.
+
+The main purpose of Polybar is to help users create awesome status bars. It has built-in functionality to display information about the most commonly used services.
 
 %prep
-%setup -q
+%setup -q -n %{name}
+git config --global http.sslVerify false
```

Comment:
at which point in the build do you use git and why?

Reply:
Git is used in project for download google tests during build. OpenSuse leap doesn't clone git repo without ` http.sslVerify false`.
```
%check
pushd %{_target_platform}
    ctest --output-on-failure
popd
```

Reply:
OBS doesnt download anything during build because network connection is not allowed :-)

Reply:
It also means that build machine must use network. It is possibe in copr, but not in Fedora koji (dont't know about openSuse). Tests can be disabled. Another opportunity is `Source1:`, but CMakeLists have to be edited in this case.

Reply:
To have reproducible builds one needs pristine sources

Reply:
What is the rationale behind running tests when building the package?

Reply:
Maybe useful sometimes, but not really needed, i think.

Reply:
> What is the rationale behind running tests when building the package?

If one packages a python package for example it makes sense to run the tests to see if runtime behaviour also works. In the case of polybar I decided not to run tests as I think they are mostly used in the development cycle.

Reply:
I would disable tests (removing this section) here and don't run git. For the reason that build systems usually try to run in a clean environment and without internet access.

Reply:
Yeah, for polybar, tests are mainly a development tool. Also we try not to add code that breaks tests ;)